hydra-node/src/Hydra/Node/Network.hs

```diff
@@ -110,22 +110,25 @@ data NetworkConfiguration m = NetworkConfiguration
 
 -- | Starts the network layer of a node, passing configured `Network` to its continuation.
 withNetwork ::
-  (ToCBOR msg, ToJSON msg, FromJSON msg, FromCBOR msg) =>
+  forall inbound outbound tx.
+  (ToCBOR inbound, ToJSON inbound, FromCBOR inbound) =>
+  (ToCBOR outbound, ToJSON outbound, FromJSON outbound, FromCBOR outbound) =>
```

Comment:
Note that at this level there should not need to be a need to distinguish inbound and outbound messages, i.e. one type variable should suffice here (to plug in `ReqTx | ReqSn | AckSn | ..` later here)

However, the returned `NetworkComponent` will be something like: `NetworkComponent IO (Authenticated (Either ControlMessage msg)) msg ()` (or isomorphic to this)

Reply:
Something from the future / not relevant for this PR. It would correspond to information about a peer connected, disconnected or version mismatch. Basically the `Connectivity` type right now.

Reply:
Something from the future / not relevant for this PR. It would correspond to information about a peer connected, disconnected or version mismatch. Basically the Connectivity type right now.